maintainer/CI/build_cmake.sh

```diff
@@ -42,6 +42,8 @@ function end {
 
 # execute and output a command
 # handle environment variables
+[ -z "$cuda_job" ] && cuda_job="false"
+[ $cuda_job = "true" ] && [ -n "${CI_PROJECT_DIR}" ] && srcdir="${CI_PROJECT_DIR}"
```

Comment:
Why is this needed?

Reply:
`cuda_build.sh` overrides the user-provided `$srcdir` with `${CI_PROJECT_DIR}`, but if the script is not run in CI, `$srcdir` becomes an empty string, then gets the value `pwd`.
This behavior can be confusing when running the script on a local machine, so I added an extra `[ -n "${CI_PROJECT_DIR}" ]` to avoid overriding `$srcdir` when the script is not run in CI.

Reply:
Yes, why does it overwrite srcdir? Isn't that just the default?

Reply:
It probably has to do with the documentation jobs (`deploy_sphinx_documentation`, `check_sphinx`, etc.) which start with the command `cd ${CI_PROJECT_DIR}/build`. We could explicitly pass `srcdir="${CI_PROJECT_DIR}"` to the job from which the documentation jobs depend and drop the overwriting line from the build script.